lib/src/utils.dart

```diff
@@ -64,12 +90,17 @@ bool isValidPackageName(String id) =>
 
 class CamelCaseString {
   static final _camelCaseMatcher = RegExp(r'[A-Z][a-z]*');
-  static final _camelCaseTester = RegExp(r'^([_$]*)([A-Z?$]+[a-z0-9]*)+$');
+
+  // A camel case string here is defined as:
+  // * An arbitrary number of optional leading `_`s or `$`s,
+  // * followed by an upper-case letter or `?`,
+  // * followed by any number of letters, digits, `?` or `$`s.
+  static final _camelCaseTester = RegExp(r'^[_$]*[A-Z?][$?a-zA-Z\d]*$');
```

Comment:
There are unexpected `?`s in this RegExp too. Should they be `_`?

If so, the RegExp should be:
```dart
RegExp(r'^[$_A-Z][$_a-zA-Z\d]*$');
```

This RegExp is not matching the original, the original is quite hard to read. I'll update it again to fix that.